api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto

```diff
@@ -117,4 +119,10 @@ message LocalRateLimit {
   // Specifies if the local rate limit filter should include the virtual host rate limits.
   common.ratelimit.v3.VhRateLimitsOptions vh_rate_limits = 13
       [(validate.rules).enum = {defined_only: true}];
+
+  // Whether consume default token bucket when there is no matching descriptor.
```

Comment:
IIUC this change attempts to enable an "empty" token_bucket, other than those of the descriptors.
@wbpcode as code-owner, this seems to be somewhat contradicting to [this comment](https://github.com/envoyproxy/envoy/blob/cda041df9f794061b86cb9e1f16c5ee77581de5b/source/extensions/filters/http/local_ratelimit/local_ratelimit.cc#L58-L65).
Should the original intent be relaxed (i.e., allow a config with no token_bucket when descriptors are provided)?

Reply:
Sorry, but I actually didn't get you. This is a new additional flag to tell envoy to skip ratelimit **if the descriptor is not match** and `consume_default_token_bucket` is set to false.

I didn't get their relationship 🤔

Reply:
I think setting the bool value to true is essentially equivalent to having no high-level local-rate-limit [token bucket](https://github.com/envoyproxy/envoy/blob/1e6aa05cfc56385e040601211466f91978512adf/api/envoy/extensions/filters/http/local_ratelimit/v3/local_rate_limit.proto#L53), and only descriptors with their buckets, no?

Reply:
I think you mean setting it to false because the default value is true. 

If it is set to true (or not set), then nothing will be changed. The default token bucket will always be consumed.

If it is set to false, then the the default token bucket will only be consumed when all descriptors match failed.

Reply:
Yeah, sorry, I mixed up the true<->false.

> If it is set to false, then the the default token bucket will only be consumed when all descriptors match failed.

I think that setting to false, then (according to the comment), the default bucket will not be consumed at all, no?

Reply:
I think maybe both the name and comment need a update.